### PR TITLE
Fix topology free

### PR DESCRIPTION
**Pull request by plbossart**
The topology free makes no sense at the moment. The topology free is done in pcm_free(), so will be called multiple times - e.g. for each BE. It's called in error cases when it's not needed. We also have multiple logs such as warn: can't find PCM with DAI ID 5 which prevent the topology from being freed and last it's not clear if the topology free actually frees everything.

This branch is not ready to be merged but is submitted only to start discussions. We need to fix this before we upstream.

### CODE REVIEW

**Comment by ranj063** [comment]
@plbossart looks like the first commit crept in by mistake

File: sound/soc/sof/core.c [hunk 1]
```diff
@@ -361,9 +362,10 @@ static int sof_probe(struct platform_device *pdev)
 
 	return 0;
 
-comp_err:
+mach_err:
 	snd_soc_unregister_component(&pdev->dev);
-	snd_sof_free_topology(sdev);
+comp_err:
+	snd_sof_free_topology(sdev); // FIXME: is this needed
```
Comment (ranj063) [inline sound/soc/sof/core.c]: @plbossart I dont think topology is loaded at this point. So we dont need to free it yet.

**Comment by plbossart** [comment]
@ranj063 the first commit is intentional if you want to test without HDMI and without dependencies on other PRs. My hope is that *someone* [hint hint wink wink] looks into this and the PR can be replaced by a better one.

**Comment by ranj063** [comment]
@plbossart I'l look into this today. Is there any particular order in which you remove the modules?
snd_sof_pci and then snd_sof?

**Comment by plbossart** [comment]
yes, see latest on https://github.com/plbossart/soft/tree/fix/mod_scripts

File: sound/soc/sof/core.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/sof/core.c]: I removed the error handling from the update.

**Pull request merged**